MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h

```diff
@@ -57,7 +57,7 @@ class PiecewiseLinearInterpolation final
      * @return The interpolated value.
      */
     double getValue(double pnt_to_interpolate) const;
-
+	double GetCurveDerivative(double pnt_to_interpolate) const;
```

Comment:
I think `getDerivative` is sufficient, because we already dealing with a curve object...

In our [styleguide](http://ufz.github.io/styleguide/cppguide.xml) there are few important points about naming functions.